pkg/pool-stable/contracts/ComposableStablePoolRates.sol

```diff
@@ -186,21 +186,9 @@ abstract contract ComposableStablePoolRates is ComposableStablePoolStorage {
      */
     function _cacheTokenRatesIfNecessary() internal {
         uint256 totalTokens = _getTotalTokens();
-
-        // The Pool will always have at least 3 tokens so we always try to update these three caches.
-        _cacheTokenRateIfNecessary(0);
-        _cacheTokenRateIfNecessary(1);
-        _cacheTokenRateIfNecessary(2);
-
-        // Before we update the remaining caches we must check that the Pool contains enough tokens.
-        if (totalTokens == 3) return;
-        _cacheTokenRateIfNecessary(3);
-
-        if (totalTokens == 4) return;
-        _cacheTokenRateIfNecessary(4);
-
-        if (totalTokens == 5) return;
-        _cacheTokenRateIfNecessary(5);
+        for (uint256 i = 0; i < totalTokens; ++i) {
+            _cacheTokenRateIfNecessary(i);
+        }
```

Comment:
Note: These are not in the diff in #2059, but were present in the base already (applies to all these refactors in `ComposableStablePoolRates`.